www/main/components/addons/bani-controller/components/BaniController.jsx

```diff
@@ -0,0 +1,164 @@
+      // TODO: Needs to remove this DOM interaction
+      document.body.classList.remove('controller-on');
+      setListeners(false);
+      setConnection(false);
+      onEnd(code);
+      setCode('...');
+      setAdminPin('...');
+      // analytics.trackEvent('syncStopped', true);
+    } else {
+      await remoteSyncInit();
+    }
+  };
+
+  const toggleLockScreen = () => {
+    if (overlayScreen !== 'lock-screen') {
+      setOverlayScreen('lock-screen');
+    }
+  };
+
+  useEffect(() => {
+    if (canvasRef.current) {
+      syncToggle(true);
+    }
+  }, [canvasRef.current]);
+
+  useSocketListeners(isListeners, adminPin);
+
+  const baniControllerItems = getBaniControllerItems({
+    code,
+    adminPin,
+    isAdminPinVisible,
+    setAdminPinVisibility,
+    toggleLockScreen,
+  });
+
+  return (
+    <Overlay onScreenClose={onScreenClose}>
+      <div className="sync-wrapper overlay-ui ui-sync-button">
+        <div className="sync overlay-ui ui-sync-button">
+          <header className="sync-header" data-key="MOBILE_DEVICE_SYNC">
+            {title}
+          </header>
+          <div className={`sync-content-wrapper ${isFetchingCode ? 'loading' : ''}`}>
+            {/* Sync-container */}
+            <div className="sync-content">
+              {isFetchingCode ? (
+                <div className="sttm-loader" />
+              ) : (
+                <>
+                  <div className="sync-code-label">
+                    {i18n.t('TOOLBAR.SYNC_CONTROLLER.UNIQUE_CODE_LABEL')}
+                  </div>
+
+                  <div className="sync-code-num"> {code} </div>
+
+                  {baniControllerItems.map(item => (
+                    <BaniControllerItem key={item.title} {...item} />
+                  ))}
+
+                  <div className="connection-switch-container">
+                    <p>Disable all the remote connections to SikhiToTheMax</p>
+                    <Switch
+                      controlId="bani-controller"
+                      onToggle={() => {
+                        syncToggle();
+                        //           analytics.trackEvent('controller', 'connection', isConntected ? 'Enabled' : 'Disabled');
+                        //         },
+                      }}
+                      defaultValue={!isConnected}
+                    />
+                  </div>
+                </>
+              )}
+            </div>
+
+            {/* QR-container */}
+            <div className="qr-container">
+              <div className="qr-desc">{i18n.t('TOOLBAR.QR_CODE.DESC')}</div>
+              <canvas ref={canvasRef} className="qr-bani-ctr" />
+              <div className="qr-title">{i18n.t('TOOLBAR.BANI_CONTROLLER')}</div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </Overlay>
+  );
+};
+
+BaniController.propTypes = {
+  onScreenClose: PropTypes.func,
+  code: PropTypes.number,
+  adminPin: PropTypes.number,
+};
+
+export default BaniController;
```

Comment:
this could be broken up into smaller components that can go in the same folder.

Reply:
Hmm, yeah you are right. Will break these up.

Reply:
Broke up the presentational components. 

I also tried moving remoteSyncInit and SyncToggle to utils, but it didn't make much logical sense to do that. It was just increasing complexity. Let me know if you have any suggestions on how can I better modularize the code.